src/app/util/basic-types.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief The include file for all the types for the data model that are not
+ *        dependent on an individual application configuration.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * @brief Type for referring to zigbee application profile id
+ *        TODO: This is probably not needed for CHIP and should be removed.
+ *        https://github.com/project-chip/connectedhomeip/issues/3444
+ */
+typedef uint16_t EmberAfProfileId;
+
+/**
+ * @brief Type for referring to CHIP attribute id.
+ */
+typedef uint16_t CHIPAttributeId;
+typedef CHIPAttributeId EmberAfAttributeId;
+#ifdef __cplusplus
+namespace chip {
+typedef CHIPAttributeId AttributeId;
+} // namespace chip
+#endif // __cplusplus
+
+/**
+ * @brief Type for referring to CHIP cluster id.
+ */
+typedef uint16_t CHIPClusterId;
+typedef CHIPClusterId EmberAfClusterId;
+#ifdef __cplusplus
+namespace chip {
+typedef CHIPClusterId ClusterId;
+} // namespace chip
+#endif // __cplusplus
+
+/**
+ * @brief Type for referring to CHIP endpoint id.
+ */
+typedef uint8_t CHIPEndpointId;
+#ifdef __cplusplus
+namespace chip {
+typedef CHIPEndpointId EndpointId;
+} // namespace chip
```

Comment:
It may be easier to parse if we group C and C++ together rather than the typedefs themselves:

```cpp

// typedefs that are to be used by CHIP C-code
typedef uint16_t CHIPAttributeId;
typedef uint16_t CHIPClusterId;
typedef uint8_t CHIPEndpointId;

// typedefs for generate code, which expects expect Ember* prefix
typedef CHIPClusterId EmberAfClusterId;
typedef CHIPAttributeId EmberAfAttributeId;
typedef uint16_t EmberAfProfileId;

#ifdef __cplusplus
namespace chip {
typedef CHIPAttributeId AttributeId;
typedef CHIPClusterId ClusterId;
typedef CHIPEndpointId EndpointId;
}
#endif
```

seems a lot shorter.

Also wondering if the `@brief` are really needed. `CHIPClusterId` for example it is clear a 'CHIP cluster id' and the same for all the others.

Reply:
Yeah, I mostly copy/pasted the `@brief` bits.  We do need it for profile id, but I can drop it for the others.